app/src/main/java/com/sschuraytz/charcoaldrawing/Tool.java

```diff
@@ -0,0 +1,85 @@
+package com.sschuraytz.charcoaldrawing;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.PorterDuff;
+
+import java.util.Random;
+
+
+public class Tool{
+
+    private Canvas canvas;
+    private Bitmap bitmap;
+    private Paint paint;
+    private Random rand = new Random();
```

Comment:
Make this static final. That way all Tools will share it.

Reply:
> Make this static final. That way all Tools will share it.

Are you just referring to line 17 (rand variable) or to all four quoted lines?
I can't make the other variables static final without initializing them. Should I move their initialization to the same line as the declaration instead of leaving it in the constructor?

Reply:
Just rand